aws/credentials/ec2rolecreds/ec2_role_provider.go

```diff
@@ -111,6 +111,16 @@ func (m *EC2RoleProvider) Retrieve() (credentials.Value, error) {
 	}, nil
 }
 
+// GetIdentityDocument retrieves an identity document describing an instance
+func (m *EC2RoleProvider) GetIdentityDocument() (*EC2InstanceIdentityDocument, error) {
```

Comment:
I'm tempted to agree, I just need to find out how does the interface change for the user of the SDK and how more/less complicated it will be... I think it should be ok, just one extra `import`... I guess.